src/pks/transport/transport_ats_pk.cc

```diff
@@ -879,8 +898,11 @@ Transport_ATS ::AdvanceDispersionDiffusion_(double t_old, double t_new)
   if (!has_diffusion_ && !has_dispersion_) return;
   double dt = t_new - t_old;
 
-  Epetra_MultiVector& tcc_new = *S_->GetW<CompositeVector>(key_, tag_next_, passwd_)
-    .ViewComponent("cell", false);
+  // Get tcc_cvs as composite vectors (CVS). This includes both cell and face components.
+  // The face component will used to compute the diffusive mass fluxes in UpdateFlux().
+  // The tcc_new is the new tcc viewed as a cell component from the CVS
+  Teuchos::RCP<CompositeVector> tcc_cvs = S_->GetPtrW<CompositeVector>(key_, tag_next_, passwd_);
+  Epetra_MultiVector& tcc_new = *tcc_cvs->ViewComponent("cell", false);
```

Comment:
@ecoon: Can you double check to see if these pointers are correct. I got ZERO diffusive mass flux from observations which is incorrect.

Reply:
This looks good to me, but can probably also be reverted if you no longer need tcc_cvs.